results/pkg/watcher/reconciler/controller.go

```diff
@@ -59,60 +59,94 @@ func (r *reconciler) Reconcile(ctx context.Context, key string) error {
 	namespace, name, err := cache.SplitMetaNamespaceKey(key)
 	if err != nil {
 		r.logger.Errorf("invalid resource key: %s", key)
-		return nil
+		return err
 	}
 
-	// Get the Task Run resource with this namespace/name
-	tr, err := r.pipelineclientset.TektonV1beta1().TaskRuns(namespace).Get(name, metav1.GetOptions{})
-	if errors.IsNotFound(err) {
-		// The resource no longer exists, in which case we stop processing.
-		r.logger.Infof("task run %q in work queue no longer exists", key)
-		return nil
-	} else if err != nil {
-		r.logger.Errorf("Error retrieving Result %q: %s", name, err)
+	// Get the resource object with this namespace/name
+	taskRun, terr := r.pipelineclientset.TektonV1beta1().TaskRuns(namespace).Get(name, metav1.GetOptions{})
+	pipelineRun, perr := r.pipelineclientset.TektonV1beta1().PipelineRuns(namespace).Get(name, metav1.GetOptions{})
+
+	if errors.IsNotFound(terr) && errors.IsNotFound(perr) {
+		r.logger.Infof("Task/Pipeline run %q in work queue no longer exists", key)
 		return err
 	}
+	if errors.IsNotFound(perr) && terr != nil {
+		r.logger.Errorf("Error retrieving TaskRun %q: %s", name, terr)
+		return terr
+	}
+	if errors.IsNotFound(terr) && perr != nil {
+		r.logger.Errorf("Error retrieving PipelineRun %q: %s", name, perr)
+		return perr
+	}
 
-	r.logger.Infof("Receiving new Result %s/%s", namespace, tr.Name)
+	var result *pb.Result = nil
 
-	// Send the new status of the Result to the API server.
-	p, err := convert.ToProto(tr)
-	if err != nil {
-		r.logger.Errorf("Error converting to proto: %v", err)
-		return err
+	val := ""
```

Comment:
val is a bit too generic here. Is there a more descriptive name we can use?

resultID?